zerver/tests/test_user_topics.py

```diff
@@ -283,3 +283,149 @@ def set_topic_visibility_for_user(user: UserProfile, visibility_policy: int) ->
             user_profile=cordelia, visibility_policy=UserTopic.VisibilityPolicy.UNMUTED
         )[0].last_updated
         self.assertEqual(cordelia_date_unmuted, date_unmuted)
+
+    @mock.patch("zerver.actions.user_topics.send_event")
+    def test_add_unmuted_topic(self, mock_send_event: mock.MagicMock) -> None:
+        user = self.example_user("hamlet")
+        self.login_user(user)
+
+        stream = get_stream("Verona", user.realm)
+
+        url = "/api/v1/user_topics"
+
+        payloads: List[Dict[str, object]] = [
+            {
+                "stream_name": stream.name,
+                "topic": "Verona3",
+                "visibility_policy": UserTopic.VisibilityPolicy.UNMUTED,
+            },
+            {
+                "stream_id": stream.id,
+                "topic": "Verona3",
+                "visibility_policy": UserTopic.VisibilityPolicy.UNMUTED,
+            },
+        ]
+
+        mock_date_unmuted = datetime(2020, 1, 1, tzinfo=timezone.utc).timestamp()
```

Comment:
We recently added the `time_machine` module as a dependency in `test_invite.py`, so you could consider using that if it feels nicer than mocking (I've not played with it much).

Reply:
Used `time_machine` module.

(interesting module with a lot of added capabilities :))